src/workers/squadFeatureAccess.ts

```diff
@@ -0,0 +1,36 @@
+import { messageToJson, Worker } from './worker';
+import { ChangeObject } from '../types';
+import { SourceMember, Feature, FeatureType } from '../entity';
+
+interface Data {
+  sourceMember: ChangeObject<SourceMember>;
+}
+
+const worker: Worker = {
+  subscription: 'api.squad-feature-access',
+  handler: async (message, con, logger): Promise<void> => {
+    const { sourceMember: member }: Data = messageToJson(message);
+    try {
+      await con.getRepository(Feature).insert({
+        feature: FeatureType.Squad,
+        userId: member.userId,
+      });
+    } catch (err) {
+      logger.error(
+        {
+          member,
+          messageId: message.messageId,
+          err,
+        },
+        'failed to give user squad feature access',
+      );
+      // Query failed or status is duplicate
+      if (err.name === 'QueryFailedError' || err.code === 187) {
```

Comment:
If query failed I think we should throw the error. We should skip only in case of duplicate.
Also I think we defined an enum to deal with postgres error codes

Reply:
This was copied from existing implementation, but i'll take a look at the enums.
Agreed on the QueryFailed

Reply:
Ah we only got them for TypeORM errors:
```
export enum TypeOrmError {
  FOREIGN_KEY = '23503',
  DUPLICATE_ENTRY = '23505',
}
```

Reply:
We haven't migrated fully for the error code enums. It's a relatively new thing

Reply:
@rebelchris this should do it. Please use them. It should be the same code